addon/templates/search-building.hbs

```diff
@@ -1,8 +1,15 @@
-<BackRoute 
-  @backRoute="project.linked-buildings" 
-  @backModel={{@model}}
-  @backText={{t "ember-gwr.building.backToProject"}}
-/>
+{{#if @model}}
+  <BackRoute
+    @backRoute="project.linked-buildings"
+    @backModel={{@model}}
+    @backText={{t "ember-gwr.building.backToProject"}}
+  />
+{{else}}
+  <BackRoute
+    @backRoute="global-landing-page"
+    @backText={{t "ember-gwr.globalLandingPage.backToGlobal"}}
+  />
+{{/if}}
```

Comment:
I would move this logic into the `BackRoute` component. I don't think the building search or the project search should know when to show which kind of `BackRoute` component.